test/files/run/t10094.scala

```diff
@@ -0,0 +1,11 @@
+trait T[@specialized(Int) S] {
+  def initialValue: S
+  var value: S = initialValue
+}
+
+final class C[@specialized(Int) S](val initialValue: S) extends T[S]
+
+object Test {
+  def main(args: Array[String]): Unit =
+    println(new C("hi").initialValue)
```

Comment:
Worth noting that just assert saves a check file.

Reply:
TIL